keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,1006 @@
+as an extended resource using the name `deviceclass.kubernetes.io/<device-class-name>`. This implicit extended
+resource name allows the simpler API to be used for DRA resource when no special DRA features beyond those
+available via `DeviceClass` are needed.
+
+
+### Resource Claim API
+
+A special resource claim object is created to keep track of device allocations for
+extended resource. The resource claim object has the following properties:
+
+  * It is namespace scoped, like other resource claim objects.
+  * It is owned by a pod, like other resource claim objects.
+  * It has `Spec` of device.requests, with each request name being an encoding
+    of the container name and the extended resource backed by DRA name inside
+    the container. It is needed to map allocated devices back to the container
+    which requested them.
+  * Its `status.allocation.devices` and `status.allocation.reservedFor` are
+    used.
+  * It does not have annotation `resource.kubernetes.io/pod-claim-name:` as
+    it is created for the extended resource request(s) in a pod spec, not for a
+    claim in the pod spec.
+  * It does have annotation `resource.kubernetes.io/extended-resource-claim: pod-name` as
+    it is created, deleted, updated by the scheduler. It is used by scheduler
+    to find the resource claim it has created, and ensure at most one such
+    claim per pod.
+  * At most one such claim object is created per pod. For example, if a pod
+    requests for foo1.domain/bar and foo2.domain/bar, the allocation of devices
+    for each are recorded in DeviceResourceRequestAllocationResult, and just 
+    one claim object with allocation `Results` that lists all allocated devices
+    is created for the pod.
+
+The special resource claim object lifecycle is managed by the scheduler and
+garbage collector.
+
+  * It is *created* in a namespace when there is a pod with extended resource
+  request, and the extended resource is advertised by `ResourceSlice` and
+  scheduler has fit the pod to a node with the `ResourceSlice`.
+  * It is *created* by the scheduler dynamic resource plugin during
+    preBind phase. The in-memory one in the assumed cache is created earlier
+    during Reserve phase.
+  * It is *deleted*
+    * either together with the owning pod's deletion.
+    * or by the scheduler dynamic resource plugin during unReserve phase.
+    * or by the scheduler dynamic resource plugin during postFilter phase.
+  * It is *discovered* by the kubelet via `pod.Status.ExtendedResourceClaimStatus`
+  * It is *read* by the kubelet DRA device driver to prepare the devices listed
+    therein when preparing to run the pod.
+
+```go
+type DeviceRequest struct {
+	// Name can be used to reference this request in a pod.spec.containers[].resources.claims
+	// entry and in a constraint of the claim.
+	//
+	// Must be a DNS label.
+	//
+	// +required
+	Name string `json:"name" protobuf:"bytes,1,name=name"`
```

Comment:
```suggestion
	Name string
```

Same elsewhere. Please also replace tab with four spaces, for consistency and easier editing.

Reply:
done